pkg/monitoring/metrics/virt-handler/domainstats/dirty_rate_metrics_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ *
+ */
+
+package domainstats
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"kubevirt.io/client-go/api"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+
+	"kubevirt.io/kubevirt/pkg/monitoring/metrics/testing"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/stats"
+)
+
+var _ = Describe("dirty rate metrics", func() {
+	Context("on Collect", func() {
+		var (
+			vmiStats  *VirtualMachineInstanceStats
+			vmiReport *VirtualMachineInstanceReport
+		)
+		const (
+			mb = 1024 * 1024
+		)
+
+		BeforeEach(func() {
+			vmiStats = &VirtualMachineInstanceStats{
+				DomainStats: &stats.DomainStats{
+					DirtyRate: &stats.DomainStatsDirtyRate{
+						MegabytesPerSecondSet: true,
+						MegabytesPerSecond:    54,
+					},
+				},
+			}
+
+			vmiReport = newVirtualMachineInstanceReport(api.NewMinimalVMI("test-vmi"), vmiStats)
+		})
+
+		DescribeTable("should collect metrics values", func(metric operatormetrics.Metric, expectedValue float64) {
+			crs := dirtyRateMetrics{}.Collect(vmiReport)
+			Expect(crs).To(ContainElement(testing.GomegaContainsCollectorResultMatcher(metric, expectedValue)))
+		},
+			Entry("kubevirt_vmi_dirty_rate_bytes_per_second", dirtyRateBytesPerSecond, 54.0*mb),
+		)
```

Comment:
i don't think the way you made the test, it is useful to be a table, maybe you can use a simple It here, wdyt?

Reply:
Great catch!